src/app/authorization/registration/registration.component.html

```diff
@@ -1,129 +1,236 @@
-                Number is required.
-            </div>
-            <div class="form-text text-danger" *ngIf="Mobile.errors?.['pattern'] && (Mobile.dirty || Mobile.touched)">
-                Number is Invalid.
-            </div>
-            <div class="form-text text-danger" *ngIf="Mobile.errors?.['maxlength'] && (Mobile.dirty || Mobile.touched)">
-                Number is long.
-            </div>
-            <div class="form-text text-danger" *ngIf="Mobile.errors?.['minlength'] && (Mobile.dirty || Mobile.touched)">
-                Number is short.
-            </div>
+      </div>
+
+      <div class="row">
+        <div class="col-mb-3 mt-03">
+          <label for="mobile" class="form-label" style="margin-top:14px"
+            >Mobile Number</label
+          >
+          <input
+            type="number"
+            formControlName="mobile"
+            class="form-control"
+            placeholder=""
+          />
+          <div
+            class="form-text text-danger"
+            *ngIf="registrationFormValidator['mobile']?.errors?.['required'] && (registrationFormValidator['mobile']?.dirty || registrationFormValidator['mobile']?.touched)"
+          >
+            Number is required.
+          </div>
+          <div
+            class="form-text text-danger"
+            *ngIf="registrationFormValidator['mobile']?.errors?.['pattern'] && registrationFormValidator['mobile']?.touched && !registrationFormValidator['mobile']?.errors?.['required'] "
+          >
+            Please insert a valid Mobile number.
+          </div>
         </div>
-    </div>
-    <div class="row">
+      </div>
+      <div class="form-group form-check" style="margin-top: 13px">
+        <input
+          type="checkbox"
+          formControlName="acceptTerms"
+          id="acceptTerms"
+          class="form-check-input"
+          [ngClass]="{ 'is-invalid': submitted && f[acceptTerms].errors }"
+        />
+        <label
+          for="acceptTerms"
+          class="form-check-label"
+          style="margin-bottom: 5px"
+          >Accept Terms & Conditions</label
+        >
+        <div
+          *ngIf="submitted && f[acceptTerms].errors"
+          class="invalid-feedback"
+        >
+          Accept Ts & Cs is required
+        </div>
+      </div>
+      <div class="row">
         <div class="col-mb-3">
-            <div class="d-grid">
-                <button type="button" [disabled]="!registrationForm.valid" (click)="submitform()" class="btn btn-success"
-                style="margin-top: 30px;">Register Now</button>
-            </div>
+          <div class="d-grid">
+            <button
+              type="submit"
+              [disabled]="!registrationForm.valid"
+              class="btn btn-success"
+            >
+              Register Now
+            </button>
+            <p class="text-center" style="margin-top: 7px">
+              Already have an account? <a href="#">Log In</a>
```

Comment:
Wht href is used ? use routerLink instead

Reply:
I have used router link and done this change